push/src/main/java/org/aerogear/mobile/push/PushService.java

```diff
@@ -34,75 +35,103 @@
     private static final String JSON_VARIANT_ID_KEY = "variantId";
     private static final String JSON_VARIANT_SECRET_KEY = "variantSecret";
     private static final String JSON_SENDER_ID_KEY = "senderId";
 
-
     private static final List<MessageHandler> MAIN_THREAD_HANDLERS =
                     Collections.synchronizedList(new ArrayList<>());
     private static final List<MessageHandler> BACKGROUND_THREAD_HANDLERS =
                     Collections.synchronizedList(new ArrayList<>());
-    private static final String TAG = "PUSH_SERVICE";
+
+    private final static String ERROR_MESSAGE =
+                    "An error occurred while trying to load the push config";
+
+    private static SharedPreferences sharedPreferences;
+    private static MessageHandler defaultHandler;
 
     private final String deviceType = "ANDROID";
     private final String operatingSystem = "android";
     private final String osVersion = android.os.Build.VERSION.RELEASE;
 
-    private MobileCore core;
-    private String url;
-    private UnifiedPushCredentials unifiedPushCredentials;
-    private SharedPreferences sharedPreferences;
+    private final UnifiedPushCredentials unifiedPushCredentials;
+    private final Gson gson = new Gson();
 
-    private static MessageHandler defaultHandler;
+    public PushService(Context context, UnifiedPushCredentials unifiedPushCredentials) {
+        nonNull(context, "context");
+        nonNull(unifiedPushCredentials, "unifiedPushCredentials");
+
+        this.unifiedPushCredentials = unifiedPushCredentials;
+
+        initDefaultHandler(context);
+        initSharedPreference(context);
 
-    @Override
-    public String type() {
-        return "push";
     }
 
-    @Override
-    public void configure(MobileCore core, ServiceConfiguration serviceConfiguration) {
-        this.core = core;
-        this.url = serviceConfiguration.getUrl();
-        getDefaultHandler(core.getContext());
+    public static final class Builder {
 
-        try {
-            JSONObject android = new JSONObject(
-                            serviceConfiguration.getProperties().get(JSON_ANDROID_CONFIG_KEY));
+        private Context context;
+        private UnifiedPushCredentials unifiedPushCredentials = new UnifiedPushCredentials();
 
-            unifiedPushCredentials = new UnifiedPushCredentials();
-            unifiedPushCredentials.setVariant(android.getString(JSON_VARIANT_ID_KEY));
-            unifiedPushCredentials.setSecret(android.getString(JSON_VARIANT_SECRET_KEY));
-            unifiedPushCredentials.setSenderId(android.getString(JSON_SENDER_ID_KEY));
+        public Builder() {}
 
-        } catch (JSONException e) {
-            MobileCore.getLogger().error(e.getMessage(), e);
-            throw new ConfigurationNotFoundException(
-                            "An error occurred while trying to load the push config");
+        public Builder openshift() {
+
+            MobileCore mobileCore = MobileCore.getInstance();
```

Comment:
Do you think it will make sense to just wrap this method into OpenShift configuration class. Then you can have StandardConfig class etc. and use strategy pattern. It will be cleaner to maintain extend. Just hint.